src/data_provider/src/sysInfoWin.cpp

```diff
@@ -779,12 +810,29 @@ void SysInfo::getProcessesInfo(std::function<void(nlohmann::json&)> callback) co
 
 void SysInfo::getPackages(std::function<void(nlohmann::json&)> callback) const
 {
-    getPackagesFromReg(HKEY_LOCAL_MACHINE, UNINSTALL_REGISTRY, callback, KEY_WOW64_64KEY);
-    getPackagesFromReg(HKEY_LOCAL_MACHINE, UNINSTALL_REGISTRY, callback, KEY_WOW64_32KEY);
+    std::set<nlohmann::json> packages;
+
+    auto fillList
+    {
+        [&packages](nlohmann::json & data)
+        {
+            packages.insert(data);
+        }
+    };
+
+    getPackagesFromReg(HKEY_LOCAL_MACHINE, UNINSTALL_REGISTRY, fillList, KEY_WOW64_64KEY);
+    getPackagesFromReg(HKEY_LOCAL_MACHINE, UNINSTALL_REGISTRY, fillList, KEY_WOW64_32KEY);
 
     for (const auto& user : Utils::Registry{HKEY_USERS, "", KEY_READ | KEY_ENUMERATE_SUB_KEYS}.enumerate())
     {
-        getPackagesFromReg(HKEY_USERS, user + "\\" + UNINSTALL_REGISTRY, callback);
+        getPackagesFromReg(HKEY_USERS, user + "\\" + UNINSTALL_REGISTRY, fillList);
+        getStorePackages(HKEY_USERS, user, fillList);
+    }
+
+    for (auto& element : packages)
+    {
+        nlohmann::json jsPackage = element;
+        callback(jsPackage);
     }
```

Comment:
Hi @HanesSciarrone ,

I understand that using a set as the collection to prevent duplicates is a quite good chose, this way will lead to a "false callback" mechanism, because we have to get all the packages and then feed them one by one using the callback.